src/RobotInterface/YarpImplementation/src/YarpRobotControl.cpp

```diff
@@ -820,3 +828,14 @@ bool YarpRobotControl::isValid() const
 {
     return m_pimpl->robotDevice != nullptr;
 }
+
+void YarpRobotControl::getJointLimits(Eigen::VectorXd& lowerLimits,
+                                      Eigen::VectorXd& upperLimits) const
+{
+    lowerLimits.resize(m_pimpl->actuatedDOFs);
+    upperLimits.resize(m_pimpl->actuatedDOFs);
```

Comment:
Using the eigen::Ref you cannot resize it anymore so you should check that the size is correct.

Reply:
Got it. Indeed this is why I was not using eigen::Ref.

How do you suggest to do the check?

Just an assert, like:

```
assert(lowerLimits.size() == m_pimpl->actuatedDOFs)
assert(upperLimits.size() == m_pimpl->actuatedDOFs)
```

Or you prefer to change also the return type from `void` to `bool` and raise an error if size do not match?

Reply:
I would return a bool

Reply:
done with https://github.com/ami-iit/bipedal-locomotion-framework/pull/868/commits/a2809e6c8ec68a39c3c4b80b322bd4353560420a